openwisp_notifications/static/openwisp-notifications/js/notifications.js

```diff
@@ -69,30 +75,54 @@ function initNotificationDropDown($) {
     $(document).focusin(function(e){
         // Hide notification widget if focus is shifted to an element outside it
         e.stopPropagation();
-        if ($('.ow-notification-dropdown').has(e.target).length === 0){
+        if (
+            $('.ow-notification-dropdown').has(e.target).length === 0 &&
+            // Do not hide if the user is interacting with the notification dialog
+            !$('.ow-overlay-notification').is(':visible')
+        ) {
             // Don't hide if focus changes to notification bell icon
             if (e.target != $('#openwisp_notifications').get(0)) {
                 $('.ow-notification-dropdown').addClass('ow-hide');
             }
         }
     });
 
-    $('.ow-notification-dropdown').on('keyup', '*', function(e){
+    $(document).on('keyup', '*', function(e){
         e.stopPropagation();
```

Comment:
We should avoid doing such practices because it would break functionalities added by other modules. 

This is the reason for https://github.com/openwisp/openwisp-controller/issues/868 bug. 

I apologise that this exist!!!

Reply:
I think we find a way to avoid to bind to the document like that, it should be possible to specify to which element to bind, or filter the elements instead of using `*`, probably also removing `e.stopPropagation();` may be a good idea, depending if we bind this event to an HTML container shared with the rest of the UI or with some other HTML container which is specific to the notifications module.

@Dhanus3133 please let me know if the suggestion above is clear.

Reply:
Updated, let me know if that requires some changes.

Reply:
It works, thanks!